firedrake/pointquery_utils.py

```diff
@@ -197,15 +202,25 @@ def compile_coordinate_element(ufl_coordinate_element, contains_eps, parameters=
     *return_value = %(inside_predicate)s;
 }
 
-static inline void wrap_to_reference_coords(void *result_, double *x, int *return_value,
-                                            double *coords, %(IntType)s *coords_map%(extruded_arg)s, %(IntType)s cell);
+static inline void wrap_to_reference_coords(
+    void* const result_, double* const x, int* const return_value, %(IntType)s const start, %(IntType)s const end%(extruded_arg)s,
+    double const *__restrict__ coords, %(IntType)s const *__restrict__ coords_map);
 
 int to_reference_coords(void *result_, struct Function *f, int cell, double *x)
 {
     int return_value;
-    wrap_to_reference_coords(result_, x, &return_value, f->coords, f->coords_map%(nlayers)s, cell);
+    %(extr_comment_out)swrap_to_reference_coords(result_, x, &return_value, cell, cell+1, f->coords, f->coords_map);
     return return_value;
 }
+
+int to_reference_coords_xtr(void *result_, struct Function *f, int cell, int layer, double *x)
+{
+    int return_value;
+    int layers[2] = {0, layer+2};  // +2 because the layer loop goes to layers[1]-1, which is nlayers-1
+    %(non_extr_comment_out)swrap_to_reference_coords(result_, x, &return_value, cell, cell+1, layers, f->coords, f->coords_map);
```

Comment:
This is disgusting. Why do we have to comment the code out in these different cases?